### ISSUE TITLE
Castbar no longer works for targettarget

### ISSUE THREAD

**Issue by Ali1331** [oUF-wow/oUF#554]
In the recent castbar revamp for Shadowlands, the logic for enabling castbar for eventless units was broken.

Prior to the update, only the event registration was guarded by a regex match ([Old](https://github.com/oUF-wow/oUF/blob/693dde943772c7d84875e1cf16330441e5036021/elements/castbar.lua#L535)).

After the update, the entire Enable routine is blocked by the regex match ([New](https://github.com/oUF-wow/oUF/blob/d4c1657ac57fb9670ecb39fddf11c86c9d012b88/elements/castbar.lua#L379)) and so the castbar element is never registered and therefore called in UpdateAllElements.

I may have misunderstood this, apologies in that case but locally just reverting the check to be around the events seems to fix the problem. 

Thanks

**Comment by ls-** [comment]
It's done deliberately. It's impossible to use proper unit events on `*target` unit, so they're updating via a fairly heavily throttled OnUpdate, so in the past the castbar would be often out of sync and not showing accurate info because those units' states were updated every 0.5s or so. So after talking about it for a while, that castbar PR was in testing for over a year, we chose to remove support for `*target` units altogether. We probably forgot to mention it in our update guide for v10 simply due to the age of that PR.
